package/e2e/__tests__/deploy.spec.ts

```diff
@@ -1,8 +1,36 @@
 import * as path from 'path'
 import * as config from 'config'
 import { runChecklyCli } from '../run-checkly'
+import axios from 'axios'
+import Projects from '../../src/rest/projects'
+
+async function deleteProject(projectLogicalId: string) {
+  const baseURL: string = config.get('baseURL')
+  const accountId: string = config.get('accountId')
+  const apiKey: string = config.get('apiKey')
+  // Why create an axios client rather than using the one in rest/api?
+  // The rest/api client is configured based on the NODE_ENV and CLI config file, which isn't suitable for e2e tests.
+  const api = axios.create({ 
+    baseURL,
+    headers: {
+      'x-checkly-account': accountId,
+      Authorization: `Bearer ${apiKey}`,
+    },
+  })
+  const projectsApi = new Projects(api)
+  const { data: projects } = await projectsApi.getAll()
+  const project = projects.find(({ logicalId }) => logicalId === projectLogicalId)
+  if (project) {
+    await projectsApi.deleteProject(project.id)
+  }
+}
 
 describe('deploy', () => {
+  // Attempt to delete the project in case it wasn't removed during the previous run
+  beforeAll(() => deleteProject('e2e-test-deploy-project'))
```

Comment:
So we won't be able to run this in parallel

Reply:
Running the e2e tests in parallel could cause some errors.

I can limit the concurrency in the GH Action https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow. I think that this would be a nice straightforward solution. I would just limit the concurrency for e2e tests rather than for the whole PR check.

We could also support concurrent runs by using UUIDs prefixes in the project name.

I'll add the UUID prefixes.

Reply:
> I'll add the UUID prefixes.

I think that would be the best solution. We can then also have smarter cleanup script with prefix matching and created_at < 10m ago

Reply:
Sounds good, I added that to the PR